### PR TITLE
feat: support error correction and border size in qrcode tool

### PR DESCRIPTION
**Pull request by bowenliang123**
# Description

- support setting error correction and border size in qrcode tool
- change default border size from 4 to 2
- change the to label to "Generate QR Code"
- improve qrcode tool 
image
image

Fixes # (issue)

## Type of Change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update, included: [Dify Document](https://github.com/langgenius/dify-docs)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] TODO

# Suggested Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] My changes generate no new warnings
- [x] I ran `dev/reformat`(backend) and `cd web && npx lint-staged`(frontend) to appease the lint gods
- [ ] `optional` I have made corresponding changes to the documentation 
- [ ] `optional` I have added tests that prove my fix is effective or that my feature works
- [ ] `optional` New and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by Yeuoly — CHANGES_REQUESTED** [review]
this would be a better yaml configuration
```yaml
identity:
  name: qrcode_generator
  author: Bowen Liang
  label:
    en_US: Generate QR Code
    zh_Hans: 生成二维码
    pt_BR: Generate QR Code
description:
  human:
    en_US: A tool for generating QR code image
    zh_Hans: 一个用于生成二维码的工具
    pt_BR: A tool for generating QR code image
  llm: A tool for generating QR code image
parameters:
  - name: content
    type: string
    required: true
    label:
      en_US: content text for QR code
      zh_Hans: 二维码文本内容
      pt_BR: content text for QR code
    human_description:
      en_US: content text for QR code
      zh_Hans: 二维码文本内容
      pt_BR: 二维码文本内容
    form: llm
  - name: error_correction
    type: select
    required: true
    default: M
    label:
      en_US: Error Correction
      zh_Hans: 容错等级
      pt_BR: Error Correction
    human_description:
      en_US: Error Correction in L, M, Q or H, from low to high, the bigger size of generated QR code with the better error correction effect
      zh_Hans: 容错等级，可设置为低、中、偏高或高，从低到高，生成的二维码越大且容错效果越好
      pt_BR: Error Correction in L, M, Q or H, from low to high, the bigger size of generated QR code with the better error correction effect
    options:
      - value: L
        label:
          en_US: Low
          zh_Hans: 低
          pt_BR: Low
      - value: M
        label:
          en_US: Medium
          zh_Hans: 中
          pt_BR: Medium
      - value: Q
        label:
          en_US: Quartile
          zh_Hans: 偏高
          pt_BR: Quartile
      - value: H
        label:
          en_US: High
          zh_Hans: 高
          pt_BR: High
    form: form
  - name: border
    type: number
    required: true
    default: 2
    min: 0
    max: 100
    label:
      en_US: border size
      zh_Hans: 边框粗细
      pt_BR: border size
    human_description:
      en_US: border size（default to 2）
      zh_Hans: 边框粗细的格数（默认为2）
      pt_BR: border size（default to 2）
    llm: border size, default to 2
    form: form
```

**Comment by bowenliang123** [comment]
Cool, thanks for recommending the `select` type. I have forgotten this.

**Comment by bowenliang123** [comment]
Adopted your advice and tested locally. Also updated the screenshots in PR description.

File: api/core/tools/provider/builtin/qrcode/tools/qrcode_generator.py [hunk 1]
```diff
@@ -2,34 +2,69 @@
...
-        # get expression
+        # get text content
         content = tool_parameters.get('content', '')
         if not content:
             return self.create_text_message('Invalid parameter content')
 
+        # get border size
+        border = tool_parameters.get('border', '')
+        if not border or isinstance(border, str):
+            return self.create_text_message('Invalid parameter border')
```
Comment (Yeuoly) [inline api/core/tools/provider/builtin/qrcode/tools/qrcode_generator.py]: There will be an error if border is 0, maybe we should check if it's an empty string

Comment (bowenliang123) [reply api/core/tools/provider/builtin/qrcode/tools/qrcode_generator.py]: Fixed. Thanks for the carefulness.
Considering the border is a number in runtime, I would skip the type checks and make user it's in the range of 0 to 100.

**Review by Yeuoly — APPROVED** [review]

**Pull request merged**